### PR TITLE
Issue 678: Show profile name instead of user name

### PR DESCRIPTION
**Pull request by kasperg**
- Add realname module to make file
- Enable realname on update
- Add configuration to use “[forename] [surname]” as real name
- Disable Drupal truncating usernames

This will automagically show the profile name as the user name in lists and on node views.

Issue: http://platform.dandigbib.org/issues/678.

### CODE REVIEW

File: ding_staff.info [hunk 1]
```diff
@@ -61,5 +63,7 @@ features[page_manager_pages][] = ding_staff_library
 features[profile2_type][] = ding_staff_profile
 features[rules_config][] = rules_role_staff_assign
 features[taxonomy][] = ding_staff_department
+features[variable][] = realname_pattern
 features[views_view][] = ding_staff
 features[views_view][] = ding_staff_admin_users
+mtime = 1426511413
```
Comment (cableman) [inline ding_staff.info]: This mtime should not be part of the info file. It's from meta-data that drupal 7.33+ generates in the database. See https://www.drupal.org/node/2381739

File: ding_staff.module [hunk 2]
```diff
@@ -144,3 +144,15 @@ function ding_staff_og_context_handler() {
...
+
+/**
+ * Implements hook_preprocess_username().
+ */
+function ding_staff_preprocess_username(&$vars) {
+  // Drupal truncates usernames over 20 chars to 15 chars + ... by default but
+  // keeps the original name in a separate variable.
+  // Using realname module ensures that many usernames are over 20 chars long
+  // and there is no reason to truncate then so reinstitute a safe version of
+  // the original username.
```
Comment (cableman) [inline ding_staff.module:156]: Great with this comment :+1:

**Comment by cableman** [comment]
Review completed: Only one minor thing spotted.

File: ding_staff.info (hunk 1 above)
Comment (kasperg) [inline ding_staff.info]: I will remove this.

**Comment by kasperg** [comment]
@cableman: Ready for re-review.

**Comment by cableman** [comment]
Approved :-)

**Pull request merged**